core/src/main/codegen/templates/Parser.jj

```diff
@@ -4929,10 +4937,16 @@ SqlNode IntervalLiteralOrExpression() :
     (
         // literal (with quoted string)
         <QUOTED_STRING> { p = token.image; }
-        intervalQualifier = IntervalQualifier() {
-            return SqlParserUtil.parseIntervalLiteral(s.end(intervalQualifier),
-                sign, p, intervalQualifier);
-        }
+        (
+            intervalQualifier = IntervalQualifier() {
+                return SqlParserUtil.parseIntervalLiteral(s.end(intervalQualifier),
+                    sign, p, intervalQualifier);
+            }
+        |
+           {
+                return SqlParserUtil.parseSnowflakeIntervalLiteral(s.end(this), sign, p);
```

Comment:
What is `this` referring to here?

Reply:
And how come we use `this` instead of `intervalQualifier` like in the `SqlParserUtil.parseIntervalLiteral`

Reply:
`intervalQualifier` doesn't exist so we can use it. This refers to the current Interval node position.

Reply:
How come `intervalQualifier` doesn't exist in this path? It looks like it is a parameter for `IntervalLiteralOrExpression` (defined on line 4925) and the function contents go up to 4970.

Reply:
@srilman to clarify, IntervalLiteralOrExpression is declared on line 4925, but it's not initialized, so the actual value is undefined/null. It's initialized on 4941, but that occurs in a separate path.

Reply:
If you look at the generated code, this is the parser object. For example:

```Java
/**
 * SQL parser, generated from Parser.jj by JavaCC.
 *
 * <p>The public wrapper for this parser is {@link SqlParser}.
 */
public class SqlDdlParserImpl extends SqlAbstractParserImpl implements SqlDdlParserImplConstants {
```

This particular construct I fetched from elsewhere in the parser. This is the defined implementation:

```Java
/** Adds the position of the last token emitted by a parser to the list,
   * and returns a position that covers the whole range. */
  public SqlParserPos end(SqlAbstractParserImpl parser) {
    return add(parser).pos();
  }
```

So it returns the location of the last omitted token .